### PR TITLE
Merging of MMTF functionality

### PR DESCRIPTION
**Pull request by abradle**
**MMTF Reading support**
MMTF is the new binary file format for macromolecule structure data. It is still under development. More information (also under development) can be found here: http://mmtf.rcsb.org/, here: https://github.com/rcsb/mmtf and here: https://github.com/rcsb/mmtf-java
1. Added the MMTF package (org.biojava.nbio.structure.io.mmtf). Includes a decoder from MMTF and a test. 
2. Added dependency to mmtf-api and mmtf-common (alpha3) from the rcsb/mmtf project.  (https://github.com/rcsb/mmtf-java)

**Alternative location behaviour has been updated**
The following commits:
82b23d3
9730025
cecf434
0ba8656
c1ea01e
  1. All alternate location groups (except with microheterogenity) contain one copy of each atom in  the group (not just the alt loc atoms).
  2. Fixed a bug where the main group for microheterogenous groups was containing all atoms (e.g. Tryptophan would contain the aromatic nitrogens of Histidine)

**Refactoring and cleanup**
1. More changes from Compound to EntityInfo 
Following on from these:
ed563e31da46f9af5e7ec5a40bdf1453da4ada1d
c84e6ec1ee28ec4138b663497036ff2a730bcee3
2. Doclint set to none  <additionalparam>-Xdoclint:none</additionalparam> at least for the time being:
https://github.com/biojava/biojava/issues/438
3. Refactoring newS -> newStructure and removed unnecesary Exceptions

### CODE REVIEW

File: biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/SimpleMMcifConsumer.java [hunk 1]
```diff
@@ -920,6 +929,28 @@ public void documentEnd() {
...
+								if (altLocGroup.getPDBName().equals(group.getPDBName())) {
+									altLocGroup.addAtom(groupAtom);
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
```
Comment (josemduarte) [inline biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/SimpleMMcifConsumer.java]: This block is a duplicate of one above in PdbFileParser. We should extract it to some utility class.

**Comment by josemduarte** [comment]
Thanks, it looks great! Let's keep this open for a while to see if there are any more comments.

File: biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/SimpleMMcifConsumer.java (hunk 1 above)
Comment (abradle) [inline biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/SimpleMMcifConsumer.java]: 814d836
Deals with this

**Comment by heuermh** [comment]
Could these commits be rebased and squashed?  A lot of the commits could be left out via cherry pick, e.g. all the alpha release pom changes.  As far as I know those were not official releases, right?  Although they do show up in Maven Central, so what do I know.  :)

**Comment by abradle** [comment]
Hi Michael,

Yes they were official alpha-releases (as you say pushed to maven central). So I suppose it would be a shame to remove the tags at which the release were made.

Best wishes,

Anthony

**Comment by josemduarte** [comment]
I'll proceed to merge this in later today. I guess there will be some issues still to be solved but we can do that on the master branch.

**Pull request merged**